File: rust/src/models/subscription_add_on_parameterization.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::billing_period_enum::BillingPeriodEnum;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct SubscriptionAddOnParameterization {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub billing_period: Option<BillingPeriodEnum>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub committed_capacity: Option<i32>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub initial_slot_count: Option<i32>,
}

impl SubscriptionAddOnParameterization {
    pub fn new() -> Self {
        Self {
            billing_period: None,
            committed_capacity: None,
            initial_slot_count: None,
        }
    }
}

File: typescript/src/models/subscriptionAddOnParameterization.ts
// this file is @generated
import { type BillingPeriodEnum, BillingPeriodEnumSerializer } from "./billingPeriodEnum";

export interface SubscriptionAddOnParameterization {
  billingPeriod?: BillingPeriodEnum | null;

  committedCapacity?: number | null;

  initialSlotCount?: number | null;
}

export const SubscriptionAddOnParameterizationSerializer = {
  _fromJsonObject(object: any): SubscriptionAddOnParameterization {
    return {
      billingPeriod:
        object["billing_period"] != null
          ? BillingPeriodEnumSerializer._fromJsonObject(object["billing_period"])
          : undefined,
      committedCapacity: object["committed_capacity"],
      initialSlotCount: object["initial_slot_count"],
    };
  },

  _toJsonObject(self: SubscriptionAddOnParameterization): any {
    return {
      billing_period:
        self.billingPeriod != null
          ? BillingPeriodEnumSerializer._toJsonObject(self.billingPeriod)
          : undefined,
      committed_capacity: self.committedCapacity,
      initial_slot_count: self.initialSlotCount,
    };
  },
};
